packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -672,6 +672,10 @@ abstract class CustomArrayNodeBase<const T extends ImplicitAllowedTypes>
 
 	protected abstract get simpleSchema(): T;
 
+	#generationNumber: number = 0;
+	#updateGenerationNumber = getFlexNode(this).on("nodeChanged", () => {
```

Comment:
I think there are two issues here:

`#updateGenerationNumber` is undocumented, and its name is confusing. This is a function, but calling it does not update the generation number. Calling this unregisters generation number updates! Consider doing this in the constructor. If you really want to do it during property static initialization time, use a static initialization block https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Static_initialization_blocks , or save it to a property with a more clear name if you actually do need to unregister at some point (I don't think we do)

I think this might not handle unhydrated nodes correctly. Using Tree.on instead of the flex tree APis should fix that as I believe that workes correctly in this case. Ensure you have a test which creates a node thats unhydrated, then as a second step inserts it into the tree. Make sure that case still works correctly, even though the underlying flex tree dummy node gets replaced with a real one (which would break your events here)

Reply:
I have tried to use a static initialization block, but unfortunately due to how scoping works with them, I couldn't get them to work. I have also tried to create a constructor within the class and initialize it there (after calling `super()`), but it wasn't really obvious where the constructor of the inherited class was.

As for the use of flex tree APIs, I have tried using `Tree.on` which led to a cyclic dependency, so I left it as is and added testing for unhydrated nodes and checked that they passed.

Would there be any other test cases where using the flex tree apis, may lead to errors that I should add test coverage for?